x-pack/plugins/maps/public/shared/layers/sources/client_file_source/geojson_file_source.js

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { AbstractVectorSource } from '../vector_source';
+import React from 'react';
+import { GEOJSON_FILE } from '../../../../../common/constants';
+import { ClientFileCreateSourceEditor } from './create_client_file_source_editor';
+import { ESSearchSource } from '../es_search_source';
+import uuid from 'uuid/v4';
+import _ from 'lodash';
+
+export class GeojsonFileSource extends AbstractVectorSource {
+
+  static type = GEOJSON_FILE;
```

Comment:
Consider making the card for this source in the selection list a slightly different color, or at least have it stand out in some way from the other sources. The process is quite different than adding the different sources, and it would be nice to communicate this. cc @cchaos